micro_sam/sam_annotator/_widgets.py

```diff
@@ -123,6 +123,58 @@ def _add_shape_param(self, names, values, min_val, max_val, step=1):
 
         return x_param, y_param, layout
 
+    #     # New method for directory selection
+    # def _add_directory_param(self, name, value, title=None):
+    #     layout = QtWidgets.QHBoxLayout()
+    #     layout.addWidget(QtWidgets.QLabel(name if title is None else title))
+
+    #     directory_textbox = QtWidgets.QLineEdit()
+    #     directory_textbox.setText(value)
+    #     layout.addWidget(directory_textbox)
+
+    #     directory_button = QtWidgets.QPushButton("Browse")
+    #     directory_button.clicked.connect(lambda: self._get_directory(name, directory_textbox))
+    #     layout.addWidget(directory_button)
+
+    #     return layout
+    def _add_directory_param(self, name, value, title=None, select_file=False):
+        layout = QtWidgets.QHBoxLayout()
+        layout.addWidget(QtWidgets.QLabel(name if title is None else title))
+
+        directory_textbox = QtWidgets.QLineEdit()
+        directory_textbox.setText(value)
+        layout.addWidget(directory_textbox)
+
+        button_text = "Browse Directory" if not select_file else "Browse File"  # Adjust button text
+        directory_button = QtWidgets.QPushButton(button_text)
+        # Call appropriate function based on select_file
+        directory_button.clicked.connect(lambda: getattr(self, "_get_{}_path".format(
```

Comment:
It looks like this will fail, because the function is called `_get_directory`. See comment below.

Reply:
changed function name to `_get_directory_path` so it's similar to `_get_file_path`